### PR TITLE
[IA] Auto-Update rust versions in docs from registry

### PR DESCRIPTION
**Pull request by svrnm**
My goal was to create version updating similar to Java for a few other languages. I wanted to start with rust, since this is a part of our docs that is still manageable. While doing so, I figured that our [current mechanism](https://github.com/open-telemetry/opentelemetry.io/blob/main/.github/workflows/auto-update-versions.yml) will not work:

* The versions for the OTLP and stdout exporter are not available from the repos
* The tagged SDK version ([v0.21.2](https://github.com/open-telemetry/opentelemetry-rust/releases/tag/v0.21.2)) is out of sync with the version on crates.io ([v0.21.0](https://crates.io/crates/opentelemetry))

By accident I stumbled upon the fact that those version are available (and auto-updated!) via the registry now! So with this PR I propose a short code that is able to extract the version of a package from a registry entry and then show how this may work using rust as an example:

```text
opentelemetry = { version = "{{% version-from-registry otel-rust %}}", features = ["trace"] }
opentelemetry-stdout = { version = "{{% version-from-registry exporter-rust-stdout %}}", features = ["trace"] }
```

@chalin wdyt?

---

**Preview**: https://deploy-preview-3944--opentelemetry.netlify.app/docs/languages/rust/

### CODE REVIEW

**Comment by chalin** [comment]
Oh, very interesting! I'll take a closer look as soon as I can.

**Comment by theletterf** [comment]
I love that we'd leverage metadata like this!

File: content/en/docs/languages/rust/exporters.md [hunk 1]
```diff
@@ -9,11 +9,12 @@ cSpell:ignore: chrono millis ostream
 ## OTLP endpoint
 
 To send trace data to a OTLP endpoint (like the [collector](/docs/collector) or
-Jaeger) you'll want to use an exporter crate, such as `opentelemetry_otlp`:
+Jaeger) you'll want to use an exporter crate, such as
+[`opentelemetry-otlp`](https://crates.io/crates/opentelemetry-otlp):
```
Comment (chalin) [inline content/en/docs/languages/rust/exporters.md:13]:
```suggestion
[opentelemetry-otlp](https://crates.io/crates/opentelemetry-otlp):
```

File: layouts/shortcodes/version-from-registry.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            {{- . -}}
+        {{- else -}}
+            {{ errorf "No 'package.version' in registry entry %q: %s" $name $.Position -}}
+        {{- end -}}
+    {{- else -}}
+        {{ errorf "Registry entry %q not found: %s" $name $.Position -}}
+    {{- end -}}
+{{- else -}}
+  {{ errorf "Missing registry entry name: %s" $.Position -}}
+{{- end -}}
```
Comment (chalin) [inline layouts/shortcodes/version-from-registry.md:17]:
```suggestion
{{ $name := (.Get 0) -}}

{{ with $name -}}
  {{ with index $.Site.Data.registry . -}}
    {{ with .package.version -}}
      {{ . -}}
    {{ else -}}
      {{ errorf "No 'package.version' in registry entry %q: %s" $name $.Position -}}
    {{ end -}}
  {{ else -}}
    {{ errorf "Registry entry %q not found: %s" $name $.Position -}}
  {{ end -}}
{{ else -}}
  {{ errorf "Shortcode parameter 'name' is missing %s: " $.Position -}}
{{ end -}}
```

**Review by chalin — APPROVED** [review]
Very nice! See inline for some suggested edits to the new shortcode. Otherwise, LGTM ✨

**Comment by svrnm** [comment]
> Very nice! See inline for some suggested edits to the new shortcode. Otherwise, LGTM ✨

Thanks, I applied all the changes

**Pull request merged**